packages/nextjs-mf/src/plugins/container/custom-jsonp.js

```diff
@@ -0,0 +1,195 @@
+// self['webpackChunkcheckout'] = {
+//   _array: self['webpackChunkcheckout'] || [],
+//
+//   push: function () {
+//     var args = Array.prototype.slice.call(arguments);
+//     console.log('push', args);
+//     var self = this;
+//     asyncOperation(function () {
+//       Array.prototype.push.apply(self._array, args);
+//     });
+//   },
+//
+//   forEach: function (callback, thisArg) {
+//     console.log('forEach', thisArg);
+//     var self = this;
+//     var i = 0;
+//
+//     function next() {
+//       if (i < self._array.length) {
+//         asyncOperation(function () {
+//           callback.call(thisArg, self._array[i], i, self._array);
+//           i++;
+//           next();
+//         });
+//       }
+//     }
+//
+//     next();
+//   },
+// };
+
+export default `
+var chunkQueue = [];
+var resport = [];
+var cnn = '__APPNAME__'
+var chunkLoadingGlobal = self['webpackChunk' + cnn] || [];
+
+function asyncOperation() {
+  // This is just an example; replace it with your own logic.
+  return new Promise((resolve) => {
+      if(__webpack_require__.S['default']) {
+          resolve()
+      }
+      console.log('init operation completed');
+      resolve(__webpack_require__.I('default', []));
+  })
+    .then(async (r) => {
+      const libKeys = Object.keys(__webpack_require__.S.default).reverse();
+      // Load 'react' and 'next' dependencies first
+      for (const libKey of libKeys) {
+        if (!(libKey.startsWith('react') || libKey.startsWith('next'))) continue;
+        const lib = __webpack_require__.S.default[libKey];
+        const versionKeys = [Object.keys(lib).reverse()[0]]
+        for (const versionKey of versionKeys) {
+          const version = lib[versionKey];
+          if (!version.loaded) {
+            // console.log('loading', libKey, versionKey, r);
+           version.loaded = 1;
+
+            await version
+              .get()
+              .then((f) => {
+              //  console.log('loaded', libKey, versionKey, r, f());
+                version.get = () => f;
+                version.loaded = true;
+              });
+          }
+        }
+      }
+
+      // Load the rest of the dependencies
```

Comment:
refactor all this. pull share module list from chunkGraph and only load eagerly marked modules, if singleton - prefer host if possible